src/api/search_transactions.rs

```diff
@@ -0,0 +1,254 @@
+      transaction_identifier: Box::new(TransactionIdentifier::new(self.hash)),
+      operations,
+      related_transactions: None,
+      metadata: match decoded_memo.as_str() {
+        "" => None,
+        _ => Some(json!({ "memo": decoded_memo })),
+      },
+    };
+    BlockTransaction::new(block_identifier, transaction)
+  }
+}
+
+impl MinaMesh {
+  pub async fn search_transactions(
+    &self,
+    req: SearchTransactionsRequest,
+  ) -> Result<SearchTransactionsResponse, MinaMeshError> {
+    let user_commands = self.fetch_user_commands(&req).await?;
+    let user_commands_len = user_commands.len();
+    let next_offset = req.offset.unwrap_or(0) + user_commands_len as i64;
+
+    // Extract the total count from the first user command, or default to 0
+    let user_commands_total_count = user_commands.first().and_then(|uc| uc.total_count).unwrap_or(0);
+
+    // Map user commands into block transactions
+    let user_commands_bt = user_commands.into_iter().map(|uc| uc.into_block_transaction()).collect();
+
+    let response = SearchTransactionsResponse {
+      transactions: user_commands_bt,
+      total_count: user_commands_total_count,
+      next_offset: match next_offset {
+        offset if offset < user_commands_total_count => Some(offset),
+        _ => None,
+      },
+    };
+
+    Ok(response)
+  }
+
+  pub async fn fetch_user_commands(&self, req: &SearchTransactionsRequest) -> Result<Vec<UserCommand>, MinaMeshError> {
+    let max_block = req.max_block;
+    let txn_hash = req.transaction_identifier.as_ref().map(|t| &t.hash);
+    let account_identifier = req.account_identifier.as_ref().map(|a| &a.address);
+    let token_id = req.account_identifier.as_ref().and_then(|a| a.metadata.as_ref().map(|meta| meta.to_string()));
+    let status = match req.status.as_deref() {
+      Some("applied") => Some(TransactionStatus::Applied),
+      Some("failed") => Some(TransactionStatus::Failed),
+      Some(_other) => None,
+      None => None,
+    };
+    let success_status = match req.success {
+      Some(true) => Some(TransactionStatus::Applied),
+      Some(false) => Some(TransactionStatus::Failed),
+      None => None,
+    };
+    let address = req.address.as_ref();
+    let limit = req.limit.unwrap_or(100);
+    let offset = req.offset.unwrap_or(0);
+
+    let user_commands = sqlx::query_file_as!(
+      UserCommand,
+      "sql/indexer_user_commands.sql",
+      max_block,
+      txn_hash,
+      account_identifier,
+      token_id,
+      status as Option<TransactionStatus>,
+      success_status as Option<TransactionStatus>,
+      address,
+      limit,
+      offset,
+    )
+    .fetch_all(&self.pg_pool)
+    .await;
+
+    match user_commands {
+      Ok(user_commands) => Ok(user_commands),
+      Err(e) => {
+        tracing::error!("Failed to fetch user commands: {:?}", e);
+        Err(MinaMeshError::Sql(e.to_string()))
+      }
+    }
+  }
+
+  #[allow(dead_code)]
+  async fn fetch_internal_commands(
```

Comment:
What's the game plan re. these two fns?

Reply:
The plan is to implement them and concatenate results with `fetch_user_commands` result which would be the ultimate response for `search_transactions`.